wavedump/src/analyze-waveforms

```diff
@@ -90,134 +83,284 @@ def get_bins(x):
+        # in the main HDF5 group and not on a per channel basis
+        x = f.attrs['xorg'] + np.linspace(0,f.attrs['xinc']*f.attrs['points'],int(f.attrs['points']))
+
+        if ':WAVeform:FORMat' in dict(f['settings'].attrs) and f['settings'].attrs[':WAVeform:FORMat'] != 'ASC':
+            # convert word values -> voltages if the data was saved in a non-ascii format
+            y = f[channel][start:stop]*f.attrs['yinc'] + f.attrs['yorg']
+        else:
+            y = f[channel][start:stop]
+    return x*1e9, y
+
+def low_filter_SPE(time, y):
+    """
+    Returns `y` through a low pass filter. Edit the cutoff frequency by 
+    editing the function itself. `time` is just the temporal spacing between
+    the samples in `y`.
+    """
+    filter_order = 2
+    nyquist = (0.5 * (time[1] - time[0]))**(-1)
+    cutoff = 5**(-1)
+    b, a = signal.butter(filter_order, min(1, cutoff/nyquist), btype='lowpass', output='ba')
+    filter_data = signal.lfilter(b, a, y)
+    return filter_data
+
+def high_filter_SPE(time, y):
+    """
+    Returns `y` through a high pass filter. Edit the cutoff frequency by 
+    editing the function itself. `time` is just the temporal spacing between
+    the samples in `y`.
+    """
+    filter_order = 2
+    nyquist = (0.5 * (time[1] - time[0]))**(-1)
+    cutoff = 5**(-1)  # Cut off frequency for the filter measured in inverse nanoseconds
+    b, a = signal.butter(filter_order, min(1, cutoff/nyquist), btype='highpass', output='ba')
+    filter_data = signal.lfilter(b, a, y)
+    return filter_data
+
 if __name__ == '__main__':
     from argparse import ArgumentParser
     import ROOT
     import matplotlib.pyplot as plt
-    import psycopg2
-    import psycopg2.extensions
-    import os
-    import sys
 
-    parser = ArgumentParser(description='Analyze data from the Agilent scope')
-    parser.add_argument('filename',help='input filename (hdf5 format)')
+    parser = ArgumentParser(description='Analyze SPE and 511 charges')
+    parser.add_argument('filenames',nargs='+',help='input filenames (hdf5 format)')
     parser.add_argument('-o','--output', default=None, help='output file name', required=True)
+    parser.add_argument('--sodium', default=False, action='store_true', help='flag to indicate data is from a sodium source')
+    # parser.add_argument('--laser', default=False, action='store_true', help='flag to indicate data is from a laser source')
     parser.add_argument('--plot', default=False, action='store_true', help='plot the waveforms and charge integral')
-    parser.add_argument('-u','--upload', default=False, action='store_true', help='upload results to the database')
-    args = parser.parse_args()
+    parser.add_argument('--chunks', default=10000, type=int, help='number of waveforms to process at a time')
+    parser.add_argument('--pdf', default=False, action='store_true', help='prints pdfs')
+    parser.add_argument('--IT', default=300, type=float, help='SPE integration length')
```

Comment:
minor nit, but I generally don't like having to type capital letters, so I would just make this lowercase. Might also be nice to have a long and short version, i.e. ```parser.add_argument("-t","--integration-time",default=300,type=float,help="SPE integration length (ns)")```